File: Exercism Codes/JavaScript/series.js
export class Series {
  constructor(series = "") {
    this.series = series;
  }

  slices(sliceLength) {
    if (this.series.length === 0) {
      throw new Error('series cannot be empty');
    }
    if (sliceLength === 0) {
      throw new Error('slice length cannot be zero');
    }
    if (sliceLength < 0) {
      throw new Error('slice length cannot be negative');
    }
    if (sliceLength > this.series.length) {
      throw new Error('slice length cannot be greater than series length');
    }

    const result = [];
    for (let i = 0; i <= this.series.length - sliceLength; i++) {
      result.push(this.series
        .substring(i, i + sliceLength)
        .split("").map(x => parseInt(x)));
    }
    return result;
  }
}

File: Exercism Codes/Python/series.py
def slices(series, length):
    if length == 0:
        raise ValueError("slice length cannot be zero")
    if length < 0:
        raise ValueError("slice length cannot be negative")
    if len(series) == 0:
        raise ValueError("series cannot be empty")
    if length > len(series):
        raise ValueError("slice length cannot be greater than series length")

    return [series[x:x + length] for x in range(1 + (len(series) - length))]
